storage_service/locations/models/space.py

```diff
@@ -143,16 +143,31 @@ def get_child_space(self):
         return protocol_space
 
     def browse(self, path, *args, **kwargs):
-        """ Returns {'directories': [directory], 'entries': [entries]} at path.
-
-        `path` is a full path in this space.
-
-        'directories' in the return dict is the name of all the directories
-            located at that path
-        'entries' in the return dict is the name of any file (directory or other)
-            located at that path
-
-        If not implemented in the child space, looks locally.
+        """
+        Return information about the objects (files, directories) at `path`.
+
+        Attempts to call the child space's implementation.  If not found, falls
+        back to looking for the path locally.
+
+        Returns a dictionary with keys 'entries', 'directories' and 'properties'.
+
+        'entries' is a list of strings, one for each entry in that directory, both file-like and folder-like.
+        'directories' is a list of strings for each folder-like entry. Each entry should also be listed in 'entries'.
+        'properties' is a dictionary that may contain additional information for the entries.  Keys are the entry name found in 'entries', values are a dictionary containing extra information. 'properties' may not contain all values from 'entries'.
```

Comment:
I had anticipated 'properties' being more free-form than that, since not all Spaces will provide all properties (eg Duracloud doesn't do file size though LocalFS does, so I'm hesitant to provide a definitive list. The consuming code should handle if the data isn't there.  Do you still think it'd be helpful to provide a partial list with caveats?